File: frontend/mobile/lib/models/requests/create_payment_intent_request.dart
import 'purchase_request.dart';

/// Prices the hold server-side and creates the payment object the buyer confirms.
/// Mirrors `CreatePaymentIntentRequestValidator` field-for-field.
class CreatePaymentIntentRequest {
  final String holdId;
  final List<PurchaseLineItemRequest> lineItems;

  const CreatePaymentIntentRequest({required this.holdId, required this.lineItems});

  Map<String, dynamic> toJson() => {
        'holdId': holdId,
        'lineItems': lineItems.map((item) => item.toJson()).toList(),
      };
}

File: backend/services/ticketing/eTicketing.Ticketing.Business/Purchases/Dtos/CreatePaymentIntentRequest.cs
namespace eTicketing.Ticketing.Business.Purchases;

/// <summary>Prices the held reservation server-side and asks eTicketing.Payment for the intent the
/// buyer will confirm. Runs exactly the same validation POST /purchases does, so an expired hold or
/// a bad line item is caught before any payment object exists.</summary>
public record CreatePaymentIntentRequest
{
    public string HoldId { get; init; } = string.Empty;
    public IReadOnlyList<PurchaseLineItemRequest> LineItems { get; init; } = [];
}
